dagshub/common/cli.py

```diff
@@ -52,6 +55,106 @@ def mount(ctx, verbose, **kwargs):
     mount(**kwargs)
 
 
+@cli.command()
+@click.option("--repo_name", help="The repository name to set up")
+@click.option("--repo_owner", help="The organization under which the repository needs to be set up")
+@click.option("--url", help="DagsHub remote url; either provide --url or repo_name and repo_owner")
+@click.option("--host", default=config.DEFAULT_HOST, help="DagsHub instance to which you want to login")
+@click.option("--mlflow", default=True, is_flag=True, help="Set up MLFlow environment with DagsHub remote")
+@click.option("--dvc", default=False, is_flag=True, help="Set up DVC with DagsHub remote")
+@click.pass_context
+def init(ctx, repo_name, repo_owner, url, host, mlflow, dvc):
+    _init(repo_name, repo_owner, url, host, mlflow, dvc)
+
+
+def _init(repo_name=None, repo_owner=None, url=None, root=None,
+          host=config.DEFAULT_HOST, mlflow=True, dvc=False):
+    from ..upload.wrapper import create_repo
+    import urllib
+
+    # Setup required variables
+    root = root or get_project_root(Path(os.path.abspath('.')))
+
+    if url and (repo_name or repo_owner):
+        repo_name, repo_owner = None, None
+
+    if not url:
+        if repo_name is not None and repo_owner is not None:
+            url = urllib.parse.urljoin(host, f'{repo_owner}/{repo_name}')
+        else:
+            with open(root / '.git' / 'config', 'r') as cfg:
+                for line in cfg.readlines():
+                    if host in line:
+                        url = line.split()[-1][:-4]
+                        break
+    elif url[-4] == '.':
+        url = url[:-4]
+
+    if not (repo_name and repo_owner):
+        splitter = lambda x: (x[-1], x[-2]) # noqa E721
+        repo_name, repo_owner = splitter(url.split('/'))
+
+    if None in [repo_name, repo_owner, url]:
+        raise ValueError('Host not found; please specify the remote url with --url')
+
+    # Setup authentication
+    username = config.username
+    password = config.password
+    if username is not None and password is not None:
+        auth = username, password
+    else:
+        token = config.token or dagshub.auth.get_token()
+        if token is not None:
+            auth = token, token
+            bearer = HTTPBearerAuth(token)
+
+    # Configure repository
+    res = http_request("GET", urllib.parse.urljoin(host, config.REPO_INFO_URL.format(
+        owner=repo_owner,
+        reponame=repo_name)), auth=bearer or auth)
+    if res.status_code == 404:
+        create_repo(repo_name)
+
+    # Configure MLFlow
+    if mlflow:
+        os.environ['MLFLOW_TRACKING_URI'] = f'{url}.mlflow'
+        os.environ['MLFLOW_TRACKING_USERNAME'] = auth[0]
+        os.environ['MLFLOW_TRACKING_PASSWORD'] = auth[1]
```

Comment:
This will not persist for the console session and will only work for the duration of the python script.
I talked about it with Nir when were considering doing a `dagshub init` as a CLI command, and that was one of my concerns.
It's preeeetty hard to export env vars from code like that. You can look at what conda and venv are doing, but this is not a path I think we want to go down, and instead keep it leaner and have it as an init command that the user calls IN the python script before running their own code. [WandB integration with PyCaret is a good example](https://wandb.ai/cayush/customer_churn/reports/Predict-Customer-Churn-using-PyCaret-and-W-B--VmlldzoxNzM4MjA5#setup)

Reply:
Should I remove init from the CLI entirely, or keep it for DVC and repo creation?

Reply:
I don't see much point in the init function in the CLI to be honest

So we have three things that it does:
1) Create a repo if it doesn't exist on dagshub
2) Init dvc installation
3) MLFlow

And for these we have basically:
1) This one is already served by the create repo function. It can even clone the repo after creation
2) Already served by dvc init, but I can see at least some value there in terms of setting up the dvc remote
3) Doesn't work

So the only thing I'd really expose in the CLI is dvc init, but with maybe a different name. Maybe something like `dagshub dvc set-remote`, but this is up for you to decide